src/main/java/org/springframework/data/couchbase/core/CouchbaseTemplate.java

```diff
@@ -107,13 +106,25 @@ public CouchbaseTemplate(final CouchbaseClientFactory clientFactory, final Couch
 	}
 
 	@Override
+	@Deprecated
 	public ExecutableRemoveById removeById() {
-		return new ExecutableRemoveByIdOperationSupport(this).removeById();
+		return removeById(null);
```

Comment:
I assume the domainType can be null in the operation supports?

Reply:
Yes. If provided, the domainType would allow accessing annotations on the entity class (such as the scope). Without, the scope would be determined by other means (possibly an annotation from the repository interface, or the scope of the clientFactory.